pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -151,9 +151,12 @@ func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s ku
 	switch obj := obj.(type) {
 	case *v1beta1.Pipeline:
 		// Verify the Pipeline once we fetch from the remote resolution, mutating, validation and conversion of the pipeline should happen after the verification, since signatures are based on the remote pipeline contents
-		err := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
-		if err != nil {
-			return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), err)
+		vr := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
+		if vr.VerificationResultType == trustedresources.VerificationError {
+			if vr.Err != nil {
+				return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), vr.Err)
+			}
+			return nil, fmt.Errorf("remote Pipeline verification failed for object %s", obj.GetName())
```

Comment:
We also returned error in WarnNoMatchPolicy case. Just curious: shouldn't we propagate that as well instead of cutting it off here?

Reply:
WarnNoMatchPolicy case will get VerificationWarn type of VerificationResult right? So we don't consider that to be an error.